src/utilities/renderPopup.js

```diff
@@ -0,0 +1,40 @@
+import generatePopupDOM from './popupDOM.js';
+import listener from './listener.js';
+import { addNewReservation } from './postData.js';
+import { getAllReservations } from './fetchData.js';
+import { renderReservations } from './renderlist.js';
+import count from '../counterFunctions/reservationCounter.js';
+
+let counter = 0;
+let reservations = [];
+const url = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/kUJtIKt0WlDGnehZIL7s/reservations';
+const getReservationsURL = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/kUJtIKt0WlDGnehZIL7s/reservations?item_id=';
+
+const reservationPopup = (obj) => {
+  const Popup = document.createElement('div');
+  generatePopupDOM(Popup, obj);
+  const container = document.createElement('div');
+  container.classList.add('popup-container');
+  container.append(Popup);
+  const closeBtn = Popup.querySelector('.close-btn');
+  const reserve = Popup.querySelector('.reserve');
+  const userName = Popup.querySelector('.name-input');
+  const dateStart = Popup.querySelector('.startdate-input');
+  const dateEnd = Popup.querySelector('.enddate-input');
+
+  getAllReservations(obj.id, getReservationsURL)
+    .then((res) => res.json())
+    .then((data) => {
+      reservations = data;
+      counter = count(data) || 0;
+      renderReservations(reservations, counter);
+    }).catch((error) => console.log(error));
```

Comment:
- [ ] for javascript best practices try always to not commit `console.log`  to your repo. It’s ugly, it kills performance and it can make confidential data be visible to anyone using the browser tools to look at your website. kindly remove it and you can see more details in [here](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) 🙏